api/src/controllers/server.ts

```diff
@@ -83,4 +85,37 @@ router.get(
 	respond,
 );
 
+router.post(
+	'/setup',
+	asyncHandler(async (req, _res, next) => {
+		const serverService = new ServerService({ schema: req.schema });
+
+		if (await serverService.setupComplete()) {
+			throw new ForbiddenError();
+		}
+
+		try {
+			await createAdmin(req.schema, {
+				email: req.body.email,
+				password: req.body.password,
+				first_name: req.body.first_name,
+				last_name: req.body.last_name,
+			});
```

Comment:
Creating an admin account without authentication feels like it could become a security issue, allowing anyone that hits the API first to get admin access to your public instance. 
For example on cloud (or any publicly reachable instance during setup) this creates a sort of race condition where your instance is open for anyone on the internet to claim admin until you fully complete and save the setup form. 😬